src/flow-worker.c

```diff
@@ -620,29 +620,32 @@ static TmEcode FlowWorker(ThreadVars *tv, Packet *p, void *data)
                 StreamTcpSessionCleanup(p->flow->protoctx);
             }
         } else if (p->proto == IPPROTO_TCP && p->flow->protoctx) {
-            FramesPrune(p->flow, p);
-            FLOWWORKER_PROFILING_START(p, PROFILE_FLOWWORKER_TCPPRUNE);
-            StreamTcpPruneSession(p->flow, p->flowflags & FLOW_PKT_TOSERVER ?
-                    STREAM_TOSERVER : STREAM_TOCLIENT);
-            FLOWWORKER_PROFILING_END(p, PROFILE_FLOWWORKER_TCPPRUNE);
+            if (p->flags & PKT_STREAM_EST) {
```

Comment:
think this can be merged into the check on line 622 for slightly cleaner code

Reply:
nice